apps/andi/lib/andi/input_schemas/datasets/dataset.ex

```diff
@@ -77,18 +107,33 @@ defmodule Andi.InputSchemas.Datasets.Dataset do
   def validate_unique_system_name(%{changes: %{technical: technical}} = changeset) do
     id = Ecto.Changeset.get_field(changeset, :id)
     data_name = Ecto.Changeset.get_change(technical, :dataName)
-    org_name = Ecto.Changeset.get_change(technical, :orgName)
+    org_id = Ecto.Changeset.get_field(changeset, :organization_id)
+
+    case is_nil(org_id) do
+      false ->
+        org = Andi.InputSchemas.Organizations.get(org_id)
+        technical_changeset = check_uniqueness(technical, id, data_name, org.orgName)
+        Ecto.Changeset.put_change(changeset, :technical, technical_changeset)
 
-    technical_changeset = check_uniqueness(technical, id, data_name, org_name)
-    Ecto.Changeset.put_change(changeset, :technical, technical_changeset)
+      _ ->
+        technical_changeset = check_uniqueness(technical, id, data_name, nil)
+        Ecto.Changeset.put_change(changeset, :technical, technical_changeset)
+    end
   end
 
   def validate_unique_system_name(changeset) do
     id = Ecto.Changeset.get_field(changeset, :datasetId)
     data_name = Ecto.Changeset.get_change(changeset, :dataName)
-    org_name = Ecto.Changeset.get_change(changeset, :orgName)
+    org_id = Ecto.Changeset.get_field(changeset, :organization_id)
 
-    check_uniqueness(changeset, id, data_name, org_name)
+    case is_nil(org_id) do
```

Comment:
Ditto. A common "get orgName if it exists" method might be helpful here

Reply:
Fixed!